packages/flutter/lib/src/material/switch_list_tile.dart

```diff
@@ -90,7 +92,36 @@ class SwitchListTile extends StatelessWidget {
     this.dense,
     this.secondary,
     this.selected = false,
-  }) : assert(value != null),
+  }) : _switchListTileType = _SwitchListTileType.material,
+       assert(value != null),
+       assert(isThreeLine != null),
+       assert(!isThreeLine || subtitle != null),
+       assert(selected != null),
+       super(key: key);
+
+  /// Creates the wrapped switch with [Switch.adaptive].
```

Comment:
I think you want a comment like this here:
```
  /// Creates a [CupertinoSwitch] if the target platform is iOS, creates a
  /// material design switch otherwise.
```